x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/index_data_visualizer_view/index_data_visualizer_view.tsx

```diff
@@ -155,7 +155,8 @@ export const IndexDataVisualizerView: FC<IndexDataVisualizerViewProps> = (dataVi
     if (dataVisualizerProps?.currentSavedSearch !== undefined) {
       setCurrentSavedSearch(dataVisualizerProps?.currentSavedSearch);
     }
-  }, [dataVisualizerProps?.currentSavedSearch]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dataVisualizerProps?.currentSavedSearch?.id]);
```

Comment:
❓ Is the reference to `dataVisualizerProps?.currentSavedSearch` updated on each state change? did you try just? 
```suggestion
  }, [dataVisualizerProps?.currentSavedSearch?]);
```

Reply:
Updated here https://github.com/elastic/kibana/pull/136897/commits/8da69869978420cce8c30986f7f37919a6c10a9a